Objects/longobject.c

```diff
@@ -359,9 +359,13 @@ PyLong_FromLong(long ival)
         if (IS_SMALL_UINT(ival)) { \
             return get_small_int((sdigit)(ival)); \
         } \
+        if ((ival) <= PyLong_MASK) { \
```

Comment:
What about negative values? Should we change the test to this?

```suggestion
        if (-PyLong_MASK <= (ival) && (ival) <= PyLong_MASK) { \
```